release notes/v0.27.0.md

```diff
@@ -0,0 +1,142 @@
+
+
+## Internals
+
+- As mentioned above, [#1007](https://github.com/loadimpact/k6/issues/1007) was almost a complete re-write of the execution scheduling parts of k6. This involved deep architectural changes in how test runs are executed and improved overall code coverage by around 2%.
+- Switched to Go 1.14 for building and testing k6, bringing some fixes and performance improvements.
+- WebSockets: Updated gorilla/websocket library bringing minor performance improvements ([#1515](https://github.com/loadimpact/k6/pull/1515)).
+- Code cleanup and formatting. Thanks @thinkerou!
+
+
+## Breaking changes
+
+- Execution config options (`scenarios`, `stages`, `iterations`, `duration`) from "upper" config layers overwrite execution options from "lower" (i.e. CLI flags > environment variables > JS options > JSON options) config layers. For example, the `--iterations` CLI flag will overwrite any execution options set as environment variables (e.g. `K6_DURATION`, `K6_STAGES`, etc.) or script options (`stages: [ /* ... */]`, `scenarios: { /* ... */ }`, etc. ).
+
+- Previously, all iterations were interruptible - as soon as the specified `duration` expired, or when VUs were ramped down in a stage, any running iterations were interrupted. Now all executors besides the `externally-controlled` one have a `gracefulStop` period of `30s` by default ([#898](https://github.com/loadimpact/k6/issues/898)). Additionally, the `ramping-vus` executor has a `gracefulRampDown` parameter that configures the ramp-down grace period. For those periods, no new iterations will be started by the executors, but any currently running iterations will be allowed up to the specified periods to finish their execution.
+
+- Using different execution config options on the same level is now a configuration conflict error and will abort the script. For example, executing `k6 run --duration 10s --stages 5s:20 script.js` won't work ([#812](https://github.com/loadimpact/k6/issues/812)). The only exception is combining `duration` and `iterations`, which will result in a `shared-iterations` executor with the specified non-default `maxDuration` ([#1058](https://github.com/loadimpact/k6/issues/1058)).
+
+- The k6 REST API for controlling script execution (i.e. the `k6 pause`, `k6 scale` commands) now only works when a `externally-controlled` executor is configured in the `scenarios` config. The initial pausing of a test (i.e. `k6 run --paused script.js`) still works with all executor types, but once the test is started with `k6 resume` (or the corresponding REST API call), it can't be paused again unless only the `externally-controlled` executor is used.
+
+- Previously, running a script with `k6 run --paused script.js` would have still executed the script's `setup()` function (if it was present and wasn't explicitly disabled with `--no-setup`) and paused immediately after. Now, k6 will pause before it executes `setup()`.
+
```

Comment:
```suggestion

- Previously, if you ramped down and ramped up VUs via `stages`, the `__VU` global variables would have been incremented on the ramp-ups. This will no longer happen, the max value of `__VU` across a test run will never exceed the number of initialized VUs.

```